core/src/androidUnitTest/kotlin/com/datadog/kmp/DatadogExtTest.kt

```diff
@@ -0,0 +1,153 @@
+            UploadFrequency.FREQUENT to UploadFrequencyAndroid.FREQUENT,
+            UploadFrequency.AVERAGE to UploadFrequencyAndroid.AVERAGE,
+            UploadFrequency.RARE to UploadFrequencyAndroid.RARE
+        )
+            .assertExhaustive(UploadFrequency.entries)
+            .assertAllKeysEqualToValuesWhen { it.native }
+    }
+
+    @Test
+    fun `M return valid native value W BatchProcessingLevel_native`() {
+        mapOf(
+            BatchProcessingLevel.HIGH to BatchProcessingLevelAndroid.HIGH,
+            BatchProcessingLevel.MEDIUM to BatchProcessingLevelAndroid.MEDIUM,
+            BatchProcessingLevel.LOW to BatchProcessingLevelAndroid.LOW
+        )
+            .assertExhaustive(BatchProcessingLevel.entries)
+            .assertAllKeysEqualToValuesWhen { it.native }
+    }
+
+    @Test
+    fun `M return valid native value W BatchSize_native`() {
+        mapOf(
+            BatchSize.LARGE to BatchSizeAndroid.LARGE,
+            BatchSize.MEDIUM to BatchSizeAndroid.MEDIUM,
+            BatchSize.SMALL to BatchSizeAndroid.SMALL
+        )
+            .assertExhaustive(BatchSize.entries)
+            .assertAllKeysEqualToValuesWhen { it.native }
+    }
+
+    @Test
+    fun `M return valid native value W DatadogSite_native`() {
+        mapOf(
+            DatadogSite.US1 to DatadogSiteAndroid.US1,
+            DatadogSite.US1_FED to DatadogSiteAndroid.US1_FED,
+            DatadogSite.US3 to DatadogSiteAndroid.US3,
+            DatadogSite.US5 to DatadogSiteAndroid.US5,
+            DatadogSite.EU1 to DatadogSiteAndroid.EU1,
+            DatadogSite.AP1 to DatadogSiteAndroid.AP1
+        )
+            .assertExhaustive(DatadogSite.entries)
+            .assertAllKeysEqualToValuesWhen { it.native }
+    }
+
+    @Test
+    fun `M return expected configuration W Configuration_native`(@Forgery configuration: Configuration) {
+        val coreConfig = configuration.coreConfig
+        assertThat(configuration.native)
+            .isEqualTo(
+                ConfigurationAndroid.Builder(
+                    clientToken = configuration.clientToken,
+                    env = configuration.env,
+                    variant = configuration.variant,
+                    service = configuration.service
+                )
+                    .useSite(coreConfig.site.native)
+                    .setBatchSize(coreConfig.batchSize.native)
+                    .setUploadFrequency(coreConfig.uploadFrequency.native)
+                    .setBatchProcessingLevel(coreConfig.batchProcessingLevel.native)
+                    .setCrashReportsEnabled(coreConfig.trackCrashes)
+                    .setAdditionalConfiguration(
+                        mapOf(
+                            InternalAttributes.SOURCE_ATTRIBUTE,
+                            InternalAttributes.SDK_VERSION_ATTRIBUTE
+                        )
+                    )
+                    .setProxy(coreConfig.proxyConfiguration)
+                    .build()
```

Comment:
minor: better to create `expectedConfiguration` in the `// Given` block and use it in the `// Then` block to lighten it up.